File: Python/boj/Silver/2910.py
import sys
from collections import Counter

if __name__ == "__main__":
    try:
        sys.stdin = open("input.txt", "r")
    except FileNotFoundError:
        pass

    N, C = map(int, input().split())
    numbers = list(map(int, input().split()))
    counter = Counter(numbers)
    result = []

    sortedCount = sorted(counter.items(), key=lambda x: x[1], reverse=True)

    for n, c in sortedCount:
        for _ in range(c):
            result.append(n)

    print(" ".join(map(str, result)))

File: JavaScript/boj/Silver/2910.js
const fs = require('fs');
const filePath = process.platform === 'linux' ? '/dev/stdin' : './input.txt';
const input = fs.readFileSync(filePath).toString().trim().split('\n');

const [N, C] = input[0].split(' ').map(Number);
const numbers = input[1].split(' ').map(Number);
const count = new Map();
const result = [];

for (let i = 0; i < N; i++) {
  if (count.has(numbers[i])) count.set(numbers[i], count.get(numbers[i]) + 1);
  else count.set(numbers[i], 1);
}

const sortedCount = [...count.entries()].sort((a, b) => b[1] - a[1]);

sortedCount.forEach(([n, c]) => {
  for (let i = 0; i < c; i++) result.push(n);
});

console.log(result.join(' '));
